pkg/store/bucket.go

```diff
@@ -1691,7 +1691,9 @@ func (s *BucketStore) Series(req *storepb.SeriesRequest, seriesSrv storepb.Store
 	// Merge the sub-results from each selected block.
 	tracing.DoInSpan(ctx, "bucket_store_merge_all", func(ctx context.Context) {
 		begin := time.Now()
-		set := NewResponseDeduplicator(NewProxyResponseLoserTree(respSets...))
+		lt := NewProxyResponseLoserTree(respSets...)
+		defer lt.Close()
```

Comment:
Do we need to call `Close()` outside of this `func(context.Context)` because before `Flush()` we could still hold onto these sets?

Reply:
Yes this is a good point. Is it okay if we move `lt := NewProxyResponseLoserTree(respSets...)` outside of the span function as well, and we can defer a close right after it.